plugins/tiddlywiki/multiwikiserver/modules/store/sql-tiddler-store.js

```diff
@@ -151,12 +151,15 @@ SqlTiddlerStore.prototype.processIncomingTiddler = function(tiddlerFields) {
 	const attachmentsEnabled = this.adminWiki.getTiddlerText("$:/config/MultiWikiServer/EnableAttachments","yes") === "yes";
 	const contentTypeInfo = $tw.config.contentTypeInfo[tiddlerFields.type || "text/vnd.tiddlywiki"],
 		isBinary = !!contentTypeInfo && contentTypeInfo.encoding === "base64";
-	if(attachmentsEnabled && isBinary && tiddlerFields.text && tiddlerFields.text.length > attachmentSizeLimit) {
-		const attachment_blob = this.attachmentStore.saveAttachment({
+	if(attachmentsEnabled && isBinary && ((tiddlerFields.text && tiddlerFields.text.length > attachmentSizeLimit) || (tiddlerFields?.attachment_blob && tiddlerFields?.attachment_blob?.length < attachmentSizeLimit))) {
```

Comment:
The problem here is that "attachment_blob" is not a tiddler field. It is a property that is returned by some of the methods of SqlTiddlerDatabase.

What we need to do here is (a) look the tiddler up in the database to see if it is currently saved as an attachment and (b) check if the incoming tiddler has a _canonical_uri field with the expected value. If both of those are true, then we save the tiddler fields except _canonical_uri, and maintain the same attachment.